fuel-metrics/src/core_metrics.rs

```diff
@@ -1,89 +1,73 @@
-use axum::{
-    body::Body,
-    http::header::CONTENT_TYPE,
-    response::{
-        IntoResponse,
-        Response,
-    },
-};
 use lazy_static::lazy_static;
-use prometheus::{
-    self,
-    register_int_counter,
-    Encoder,
-    IntCounter,
-    TextEncoder,
+use prometheus_client::{
+    metrics::{
+        counter::Counter,
+        histogram::Histogram,
+    },
+    registry::Registry,
 };
 
-/// DatabaseMetrics is a wrapper struct for all
-/// of the initialized counters for Database-related metrics
-#[derive(Clone, Debug)]
 pub struct DatabaseMetrics {
-    pub write_meter: IntCounter,
-    pub read_meter: IntCounter,
-    pub bytes_written_meter: IntCounter,
-    pub bytes_read_meter: IntCounter,
+    pub registry: Registry,
+    // For descriptions of each Counter, see the `new` function where each Counter/Histogram is initialized
+    pub write_meter: Counter,
+    pub read_meter: Counter,
+    pub bytes_written: Histogram,
+    pub bytes_read: Histogram,
 }
 
-lazy_static! {
-    pub static ref DATABASE_METRICS: DatabaseMetrics = DatabaseMetrics {
-        write_meter: register_int_counter!(
-            "Writes",
-            "Number of database write operations"
-        )
-        .unwrap(),
-        read_meter: register_int_counter!("Reads", "Number of database read operations")
-            .unwrap(),
-        bytes_written_meter: register_int_counter!(
-            "Bytes_Written",
-            "The number of bytes written to the database"
-        )
-        .unwrap(),
-        bytes_read_meter: register_int_counter!(
-            "Bytes_Read",
-            "The number of bytes read from the database"
-        )
-        .unwrap(),
-    };
+impl DatabaseMetrics {
+    fn new() -> Self {
+        let registry = Registry::default();
+
+        let write_meter: Counter = Counter::default();
+        let read_meter: Counter = Counter::default();
+
+        let bytes_written = Vec::new();
+        let bytes_written_histogram = Histogram::new(bytes_written.into_iter());
+
+        let bytes_read = Vec::new();
+        let bytes_read_histogram = Histogram::new(bytes_read.into_iter());
+
+        DatabaseMetrics {
+            registry,
+            write_meter,
+            read_meter,
+            bytes_read: bytes_read_histogram,
+            bytes_written: bytes_written_histogram,
+        }
+    }
 }
 
-/// CoreMetrics tracks metrics for most of fuel-core, wrapping core metrics similar to
-/// DatabaseMetrics, kept seperate from GQLMetrics which holds more specific endpoint metrics
-#[derive(Clone, Debug)]
-pub struct CoreMetrics {
```

Comment:
Hmmm, why did we remove these metrics?

Reply:
I see that we don't use them, but maybe it is time to start=D Or we can create an issue with a list of all metrics that we want to have and put this metric there(to not forget). I think it may be useful

Reply:
Yeah I can break out an issue, we have this a bit in #483 which should have everything, but if you have any suggestions please add there

Reply:
Oh, I forgot to check the main issue. It's already there=) Nevermind